ceph_deploy/gatherkeys.py

```diff
@@ -1,84 +1,262 @@
+        for line in err:
+            rlogger.debug(line)
+        return False
+    mon_status = json.loads("".join(out))
+    mon_number = None
+    mon_map = mon_status.get('monmap')
+    if mon_map is None:
+        rlogger.error("could not find mon map for mons on '%s'" % (host))
+        return False
+    mon_quorum = mon_status.get('quorum')
+    if mon_quorum is None:
+        rlogger.error("could not find quorum for mons on '%s'" % (host))
+        return False
+    for mon in mon_map.get('mons'):
+        if mon.get('name') == host:
+           mon_number = mon.get('rank')
+           break
+    if mon_number is None:
+        rlogger.error("could not find '%s' in monmap" % (host))
+        return False
+    if not mon_number in mon_quorum:
+        rlogger.error("Not yet quorum for '%s'" % (host))
+        return False
+    got_all_keys = True
+    for keytype in ["admin", "mds", "osd", "rgw"]:
+        rc = gatherkeys_missing(args, distro, rlogger, path_keytype_mon, keytype, dest_dir)
+        if not rc:
+            got_all_keys = False
+    return got_all_keys
+
+
+def gatherkeys(args):
+    """
+    Gather keys from any mon and store in current working directory.
+
+    Backs up keys from previous installs and stores new keys.
+    """
+    oldmask = os.umask(077)
+    try:
+        try:
+            tmpd = tempfile.mkdtemp()
+            LOG.info("Storing keys in temp directory %s" %(tmpd))
+            sucess = False
+            for host in args.mon:
+                sucess = gatherkeys_with_mon(args, host, tmpd)
+                if sucess:
+                    break
+            if not sucess:
+                LOG.warning("No mon key found. Is '%s' a mon node" % (host))
+            had_error = False
+            date_string = time.strftime("%Y%m%d%H%M%S")
+            for keytype in ["admin", "mds", "mon", "osd", "rgw"]:
+                filename = keytype_path_to(args, keytype)
+                tmp_path = os.path.join(tmpd, filename)
+                if not os.path.exists(tmp_path):
+                    LOG.error("No key retrived for '%s'" % (keytype))
+                    had_error = True
+                    continue
+                if not os.path.exists(filename):
+                    LOG.info("Storing %s" % (filename))
+                    shutil.move(tmp_path, filename)
+                    continue
+                if _keyring_equivilent(tmp_path, filename):
+                    LOG.info("keyring '%s' already exists" % (filename))
+                    continue
+                backup_keyring = "%s-%s" % (filename, date_string)
+                LOG.info("Replacing '%s' and backing up old key as '%s'" % (filename, date_string))
+                shutil.copy(filename, backup_keyring)
+                shutil.move(tmp_path, filename)
+            if had_error:
+                raise RuntimeError('Faild to get all key types')
+        finally:
+            LOG.info("Destroy temp directory %s" %(tmpd))
```

Comment:
in some places the code calls this "a backup" and here it is a "temp directory". Why is it copying everything to the temp directory to remove everything afterwards?

Doesn't read as a backup.